Source/SwiftLintBuiltInRules/Rules/Idiomatic/UnneededBreakInSwitchRule.swift

```diff
@@ -45,23 +45,118 @@ struct UnneededBreakInSwitchRule: SwiftSyntaxRule, ConfigurationProviderRule {
+            embedInSwitch("something()\n    ↓break\n" + docBlockComment)
+            : embedInSwitch("something()\n" + docBlockComment),
+            embedInSwitch("something()\n    ↓break", case: "default")
+            : embedInSwitch("something()", case: "default"),
+            embedInSwitch("something()\n    ↓break", case: "case .foo, .foo2 where condition")
+            : embedInSwitch("something()", case: "case .foo, .foo2 where condition")
         ]
     )
 
     func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
         UnneededBreakInSwitchRuleVisitor(viewMode: .sourceAccurate)
     }
+
+    func makeRewriter(file: SwiftLintCore.SwiftLintFile) -> SwiftLintCore.ViolationsSyntaxRewriter? {
+        UnneededBreakInSwitchRewriter(
+            locationConverter: file.locationConverter,
+            disabledRegions: disabledRegions(file: file)
+        )
+    }
 }
 
 private final class UnneededBreakInSwitchRuleVisitor: ViolationsSyntaxVisitor {
     override func visitPost(_ node: SwitchCaseSyntax) {
-        guard node.statements.count > 1,
-              let statement = node.statements.last,
-              let breakStatement = statement.item.as(BreakStmtSyntax.self),
-              breakStatement.label == nil else {
+        guard node.containsUnneededBreak(),
+              let statement = node.statements.last else {
             return
         }
 
         violations.append(statement.item.positionAfterSkippingLeadingTrivia)
     }
 }
+
+private final class UnneededBreakInSwitchRewriter: SyntaxRewriter, ViolationsSyntaxRewriter {
+    private(set) var correctionPositions: [SwiftSyntax.AbsolutePosition] = []
+    let locationConverter: SourceLocationConverter
+    let disabledRegions: [SourceRange]
+
+    init(locationConverter: SourceLocationConverter, disabledRegions: [SourceRange]) {
+        self.locationConverter = locationConverter
+        self.disabledRegions = disabledRegions
+    }
+
+    override func visit(_ node: SwitchCaseSyntax) -> SwitchCaseSyntax {
+        guard let statement = node.statements.last,
+              node.containsUnneededBreak(),
+              !node.isContainedIn(regions: disabledRegions, locationConverter: locationConverter) else {
+            return super.visit(node)
+        }
+
+        correctionPositions.append(statement.item.positionAfterSkippingLeadingTrivia)
+
+        let trivia = statement.item.leadingTrivia + statement.item.trailingTrivia
+
+        let stmts = node.statements.removingLast()
+        let secondLast = stmts.last!
+
+        let newNode = node
+            .with(\.statements, stmts)
+            .with(\.statements.trailingTrivia, secondLast.item.trailingTrivia + trivia)
+            .trimmed { !$0.isComment }
+            .formatted()
+            .as(SwitchCaseSyntax.self)!
+
+        return super.visit(newNode)
+    }
+}
+
+private extension SwitchCaseSyntax {
+    func containsUnneededBreak() -> Bool {
```

Comment:
Nit: Such functions are typically just computed properties in the SwiftLint codebase.

Returning the unneeded break statement here (or `nil` if there isn't any) would spare you the `let statement = node.statements.last` lines right after calling this function.

Reply:
Fixed in 968b32c